### PR TITLE
[FEATURE] Add Checkpoint.run analytics

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/analytics/actions.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 DATA_CONTEXT_INITIALIZED: Action = Action("data_context.initialized")
 CHECKPOINT_CREATED: Action = Action("checkpoint.created")
 CHECKPOINT_DELETED: Action = Action("checkpoint.deleted")
+CHECKPOINT_RAN: Action = Action("checkpoint.ran")
```
Comment (cdkini) [inline great_expectations/analytics/actions.py:9]: ran vs run?

File: great_expectations/analytics/events.py [hunk 2]
```diff
@@ -188,6 +189,17 @@ def __init__(self, checkpoint_id: str | None = None):
...
 
+@dataclass
+class CheckpointRanEvent(_CheckpointEvent):
+    _allowed_actions: ClassVar[List[Action]] = [CHECKPOINT_RAN]
+
+    def __init__(self, checkpoint_id: str | None = None):
+        super().__init__(
+            action=CHECKPOINT_RAN,
+            checkpoint_id=checkpoint_id,
+        )
```
Comment (cdkini) [inline great_expectations/analytics/events.py]: Do we want additional values (like the validation definition ids)? Are we aligned with Tetris on this?

File: tests/checkpoint/test_checkpoint.py [hunk 3]
```diff
@@ -589,7 +590,12 @@ def test_checkpoint_run_dependencies_not_added_raises_error(
         ]
 
     @pytest.mark.unit
-    def test_checkpoint_run_no_actions(self, validation_definition: ValidationDefinition):
+    def test_checkpoint_run_no_actions(
+        self, validation_definition: ValidationDefinition, mocker: MockerFixture
+    ):
+        context = mocker.Mock(spec=AbstractDataContext)
+        set_context(project=context)
+        mocker.patch("great_expectations.checkpoint.checkpoint.submit_analytics_event")
```
Comment (cdkini) [inline tests/checkpoint/test_checkpoint.py]: Do we have to mock this? Or can we disable usage stats such that we don't even care about this call? I'm pretty confident that CI won't send events like this

File: great_expectations/analytics/actions.py (hunk 1 above)
Comment (joshua-stauffer) [reply great_expectations/analytics/actions.py:9]: every analytics action is named in the past tense, so `checkpoint.ran` is consistent.

File: great_expectations/analytics/events.py (hunk 2 above)
Comment (joshua-stauffer) [reply great_expectations/analytics/events.py]: i'll add them 👍

File: tests/checkpoint/test_checkpoint.py (hunk 3 above)
Comment (joshua-stauffer) [reply tests/checkpoint/test_checkpoint.py]: thanks! i was just trying to do due diligence to not send events from tests, whether from CI or when triggered by a developer, but I can remove this is it isn't necessary.

**Review by cdkini — APPROVED** [review]

**Pull request merged**